lib/can/can_bit_timing.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Heavily based on code from LinCAN sources, the H8S2638 project, and the
+ * linux-can projects.
+ * Copyright 2004-2006 Pavel Pisa - DCE FELK CVUT cz
+ * Copyright 2005      Stanislav <pisa@cmp.felk.cvut.cz>
+ * Copyright 2019      Matt Walker <m@allocor.tech>
```

Comment:
yes, this will be special.  What sort of licenses do those projects have? Are you sure you can add them to an lgpl project?